jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtil.java

```diff
@@ -115,17 +115,33 @@ public static Properties getPropertiesWithSystemOverrides(JavaProject project) {
         return properties;
     }
 
+  /**
+   * Infer final output artifact of provided project. We first infer artifact manually
+   * from buildFinalName or artifactId-version instead of directly using <code>jkubeProject.getArtifact()</code>
+   * This is done to handle case of maven where value of artifact varies depending upon the maven phase
+   * in which goal was executed. You can read more about this issue here: https://github.com/eclipse/jkube/issues/817
+   *
+   * If inferred file is not found; <code>jkubeProject.getArtifact()</code> is checked, otherwise null is returned.
+   *
+   * @param jkubeProject Project which
+   * @return final output artifact file
+   */
```

Comment:
```suggestion
  /**
   * Returns the output artifact for the Project.
   * 
   * <p> This method infers the Porject's artifact from the provided project build directory, GAV coordinates,
   *  packaging, and configured build name.
   * 
   * <p> If the inferred artifact file exists it is returned.
   *
   * <p> If the inferred artifact file doesn't exist, but an existent artifact file is provided in the configuration, it is
   * returned instead.
   
   * <p> In case no artifacts are found or configured, this method returns null.
   *
   * <p> TODO: https://github.com/eclipse/jkube/issues/817
   * <br> This method prioritizes artifact inference instead of configuration. This is done to handle case of maven
   * where value of artifact varies depending upon the maven phase in which goal was executed. 
   * 
   * @see <a href="https://github.com/eclipse/jkube/issues/817">#817</a>
   *
   * @param jkubeProject project for which to retrieve the output artifact.
   * @return the final output artifact file or null if it doesn't exist.
   */
```